### PR TITLE
ekf2: estimator_status_flags

### PR DESCRIPTION
**Pull request by dagar**
Follow up to https://github.com/PX4/PX4-Autopilot/issues/16372.

This is a bit verbose and expensive in terms of flash, but it's extremely valuable to be able to monitor the status of the estimator(s) trivially. 

Currently we have duplicate definitions of these bits across PX4 msgs, ecl analysis script, flight review, and other custom tools. 

For compatibility I've left estimator_status filter_fault_flags, innovation_check_flags, and solution_status_flags in place, but they can be phased out.

TODO:
 - ~~innovation_fault_status_u~~
 - single `estimator_status_flags` message or split across `estimator_control_status`, `estimator_fault_status`, `estimator_innovation_failures`?
 - find more flash...
 - review `solution_status_flags`, this seems to be mavlink specific? https://github.com/PX4/PX4-ECL/blob/1541e4b414fe63ec9d6c3077375db7e20c20ef4a/EKF/common.h#L487-L500

### CODE REVIEW

**Comment by dagar** [comment]
Overflowed by 2137 bytes, not great, not terrible. I'll figure out how to scrounge enough flash together.

File: msg/estimator_status_flags.msg [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+bool fault_status_bad_acc_vertical        # 16 - true if bad vertical accelerometer data has been detected
+bool fault_status_bad_acc_clipping        # 17 - true if delta velocity data contains clipping (asymmetric railing)
+
+
+# innovation test failures
+uint32 innovation_fault_status_changes    # number of filter fault status changes
+bool reject_hor_vel                       #  0 - true if horizontal velocity observations have been rejected
+bool reject_ver_vel                       #  1 - true if vertical velocity observations have been rejected
+bool reject_hor_pos                       #  2 - true if horizontal position observations have been rejected
+bool reject_ver_pos                       #  3 - true if true if vertical position observations have been rejected
```
Comment (bresch) [inline msg/estimator_status_flags.msg:63]:
```suggestion
bool reject_ver_pos                       #  3 - true if vertical position observations have been rejected
```
:D

**Review by bresch — APPROVED** [review]
Thanks, I think it's fine to have all of them in the same msg

File: msg/estimator_status_flags.msg (hunk 1 above)
Comment (dagar) [reply msg/estimator_status_flags.msg]: Thanks, copied verbatim from ecl if that wasn't obvious. https://github.com/PX4/PX4-ECL/blob/9d22f140dd6f5efc2b873a7827588f1fa8d67074/EKF/common.h#L420

**Comment by dagar** [comment]
Thanks for the quick review. I'll keep this PR ready to go as we figure out the next big flash savings in the coming days - weeks.

**Pull request merged**